core/typedefs.h

```diff
@@ -222,6 +222,12 @@ constexpr T get_num_bits(T x) {
 	return floor_log2(x);
 }
 
+// Returns `true` if a positive integer is a power of 2, `false` otherwise.
+template <typename T>
+inline bool is_power_of_2(const T x) {
+	return x && ((x & (x - 1)) == 0);
+}
+
```

Comment:
I'd move this further up next to the other `power_of_2` methods (probably as the first one).

This can be refactored to use `is_power_of_2`:
```
bool Image::is_size_po2() const {
        return uint32_t(width) == next_power_of_2(width) && uint32_t(height) == next_power_of_2(height);
}
```

And `core/os/memory.cpp` already had its own `is_power_of_2` which should be replaced by this one.

I also wonder whether this is going to raise warnings when used with unsigned types (the one in `core/os/memory.cpp` for `size_t` specifically uses unsigned literals:
```
inline bool is_power_of_2(size_t x) {
        return x && ((x & (x - 1U)) == 0U);
}
```

CC @hpvb

Reply:
Would it make it more explicit to write it as `x && ((x & (x - T(1))) == T(0));`?